release/models/interfaces/openconfig-if-ip.yang

```diff
@@ -605,6 +620,16 @@ revision "2023-06-30" {
         "RFC 4862: IPv6 Stateless Address Autoconfiguration";
     }
 
+    leaf unsolicited-na-accepted {
+      type boolean;
+      default true;
+      description
+        "When set to true unsolicited neighbor advertisements
+        will be accepted and the ARP table will be updated.";
```

Comment:
Please remove:
* any references to ARP (it doesn't exist in ipv6).
* any references to `accepted` (without specifying the conditions) - this is not what RFC9131 describes.

Unsolicited NA should be accepted with or without 9131 behavior, and will be processed in either case if the entry already exists in cache (this is why I initially asked which behavior you are trying to describe and based on your response asked for the default value).

RFC9131 changes the behavior only for the specific case when the entry doesn't exist yet, quoting the rfc:

> This document updates [[RFC4861](https://www.rfc-editor.org/rfc/rfc9131.html#RFC4861)] so that routers create a new Neighbor Cache entry upon receiving an unsolicited Neighbor Advertisement for an address that does not already have a Neighbor Cache entry. These changes do not modify the router behavior specified in [[RFC4861](https://www.rfc-editor.org/rfc/rfc9131.html#RFC4861)] for the scenario when the corresponding Neighbor Cache entry already exists.

The description should be modified accordingly

Reply:
description is updated to:
```
        When set to true neighbors will be learned from unsolicited
        neighbor advertisements.
```

Reply:
@cagdasalagoz-arista please the text to use "Neighbor Cache" instead of ARP, since this is IPv6.

Please also update the description so it is clear if you are intending
1. To enable/disable RFC4861 behavior 
or
2. To enable/disable RFC9131 behavior

The key difference between the two is whether or not the router already has a Neighbor Cache entry.  If there is already an entry, RFC4861 says it can be updated.  If there is not one, the cache won't be updated.

RFC9131 says do RFC4861 AND also update the cache is there is no entry.

As written, it is not clear in this yang node which behavior is being configured.

Reply:
Hello @dplore ,
I think you accidentally commented on the older version of the commit. The latest version is like this:
```
    leaf learn-unsolicited {
      type enumeration {
        enum NONE {
          value 0;
        }
        enum GLOBAL {
          value 1;
        }
        enum LINK_LOCAL {
          value 2;
        }
        enum BOTH {
          value 3;
        }
      }
      default "NONE";
      description
        "Sets if neighbors should be learned from unsolicited neighbor
        advertisements for global or link local addresses or both.";
      reference
        "RFC 9131: Routers Creating Cache Entries upon
        Receiving Unsolicited Neighbor Advertisements";
    }
```
Reference contains `RCF9131`

Reply:
Thanks!